test/e2e/node_feature_discovery_test.go

```diff
@@ -747,24 +814,27 @@ func waitForNfdNodeAnnotations(cli clientset.Interface, expected map[string]stri
 
 type k8sLabels map[string]string
 
-// waitForNfdNodeLabels waits for node to be labeled as expected.
-func waitForNfdNodeLabels(cli clientset.Interface, expected map[string]k8sLabels) error {
+// checkForNfdNodeLabels waits and checks that node is labeled as expected.
+func checkForNodeLabels(cli clientset.Interface, expectedNewLabels map[string]k8sLabels, oldNodes []corev1.Node) error {
+
 	poll := func() error {
 		nodes, err := getNonControlPlaneNodes(cli)
 		if err != nil {
 			return err
 		}
 		for _, node := range nodes {
-			labels := nfdLabels(node.Labels)
-			nodeExpected, ok := expected[node.Name]
+			nodeExpected, ok := expectedNewLabels[node.Name]
 			if !ok {
 				nodeExpected = k8sLabels{}
-				if defaultExpected, ok := expected["*"]; ok {
+				if defaultExpected, ok := expectedNewLabels["*"]; ok {
 					nodeExpected = defaultExpected
 				}
 			}
-			if !cmp.Equal(nodeExpected, labels) {
-				return fmt.Errorf("node %q labels do not match expected, diff (expected vs. received): %s", node.Name, cmp.Diff(nodeExpected, labels))
+
+			oldLabels := getNodeLabels(oldNodes, node.Name)
+			newNodeLabels := mergeMaps(oldLabels, nodeExpected)
```

Comment:
not a blocker, but I wonder if we could instead just use `Copy()` from [golang.org/x/exp/maps](golang.org/x/exp/maps) ?

Reply:
@marquiz @AhmedGrati any thoughts ?

Reply:
Good point @fmuyassarov, I think we should start leveraging generics. If we do this, in addition to copy we probably want to clone too to be on the safe side

```golang
newNodeLabels := maps.Copy(maps.Clone(oldLabels), nodeExpected)
```

But not a blocker in my opinion either. It's up to you folks. @AhmedGrati if you want to do this, go ahead. If not, we can do this later – and optimize (in terms of LOC) other maps usage in NFD at the same time.